{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook outlines the following steps for the `tgviz` app.  \n",
    "\n",
    "1. Read 55 AISNPs from Kidd manuscript and save to file. \n",
    "2. Download the 1000 Genomes Project `.bcf` files.\n",
    "3. Use `cyvcf2` to write one `.vcf` file with all 1000 Genomes Project samples' genotypes for each of the 55 AISNPs.\n",
    "4. Read 128 AISNPs from the Seldin Lab manuscript and save to file.\n",
    "5. Use `cyvcf2` to write one `.vcf` file with all 1000 Genomes Project samples' genotypes for each of the 128 AISNPs.\n",
    "\n",
    "Run this notebook from the `notebooks/` directory, or you will need to change the filepaths."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "from cyvcf2 import VCF, Writer\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read 55 AISNPs from [Kidd et al.](https://www.sciencedirect.com/science/article/pii/S1872497314000039) Table 1\n",
    "\n",
    "I used `pd.read_clipboard()` and copy pasted from the link to create `data/kidd_55_AISNPs.txt`,  but I commited the file to the GitHub repository."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df55 = pd.read_csv('../data/Kidd_55_AISNPs.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dbSNP rs#</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Build 37 nt position</th>\n",
       "      <th>73-population Fst</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs3737576</td>\n",
       "      <td>1</td>\n",
       "      <td>101,709,563</td>\n",
       "      <td>0.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs7554936</td>\n",
       "      <td>1</td>\n",
       "      <td>151,122,489</td>\n",
       "      <td>0.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2814778</td>\n",
       "      <td>1</td>\n",
       "      <td>159,174,683</td>\n",
       "      <td>0.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs798443</td>\n",
       "      <td>2</td>\n",
       "      <td>7,968,275</td>\n",
       "      <td>0.34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1876482</td>\n",
       "      <td>2</td>\n",
       "      <td>17,362,568</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   dbSNP rs#  Chr Build 37 nt position  73-population Fst\n",
       "0  rs3737576    1          101,709,563               0.44\n",
       "1  rs7554936    1          151,122,489               0.39\n",
       "2  rs2814778    1          159,174,683               0.82\n",
       "3   rs798443    2            7,968,275               0.34\n",
       "4  rs1876482    2           17,362,568               0.75"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df55.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following changes were made to Table 1 from the manuscript.\n",
    "\n",
    "\n",
    "| dbSNP rs# | chr, Kidd et al. | chr, new |  \n",
    "|---|---|---|\n",
    "| rs870347 | 6 | 5 |  \n",
    "| rs192655 | 7 | 6 |  \n",
    "| rs917115 | 8 | 7 |"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download the 1000 genomes .bcfs  \n",
    "\n",
    "At the command line, type the following, **these took over an hour to download for me.**\n",
    "\n",
    "```shell\n",
    "mkdir 1kg_bcfs\n",
    "wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/bcf_files/* -P 1kg_bcfs\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write a new 1kG .vcf file with the 55 Kidd AISNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace /Users/kevin/projects/tgviz with /path/to/1kg_bcfs\n",
    "bcf_path = '/Users/kevin/projects/tgviz/1kg_bcfs/'\n",
    "\n",
    "bcf_fname = 'ALL.chr{}.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.bcf'.format(1)\n",
    "bcf = VCF(os.path.join(bcf_path, bcf_fname))\n",
    "w = Writer('../data/Kidd.55AISNP.1kG.vcf', bcf)\n",
    "\n",
    "for i, aim in df55.iterrows():\n",
    "    chrom = str(aim['Chr'])\n",
    "    pos = str(aim['Build 37 nt position'])\n",
    "    pos = pos.replace(',', '')\n",
    "    rsid = str(aim['dbSNP rs#'])\n",
    "    bcf_fname = 'ALL.chr{}.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.bcf'.format(chrom)\n",
    "    bcf = VCF(os.path.join(bcf_path, bcf_fname))\n",
    "    for variant in bcf(\"{}:{}-{}\".format(chrom, pos, pos)):\n",
    "        if variant.POS == int(pos):\n",
    "            w.write_record(variant)\n",
    "w.close(); bcf.close()   "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Now, read 128 AISNPs from the Seldin Lab manuscript"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df128 = pd.read_csv('https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3073397/bin/NIHMS65513-supplement-S2.txt', \n",
    "                    sep='\\t', skiprows=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df128.to_csv('../data/Seldin_128_AISNPs.txt', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI SNP Reference</th>\n",
       "      <th>Assay ID</th>\n",
       "      <th>Strand</th>\n",
       "      <th>VIC</th>\n",
       "      <th>FAM</th>\n",
       "      <th>Context Sequence</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Celera ID</th>\n",
       "      <th>NCBI Assembly Build Number</th>\n",
       "      <th>Location on NCBI Assembly</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>Unnamed: 11</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs731257</td>\n",
       "      <td>C_____14517_10</td>\n",
       "      <td>Forward</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>AAGTTGCAATATGGCAAAACCTGTA[A/G]GAGATACAATTTGTGA...</td>\n",
       "      <td>7</td>\n",
       "      <td>hCV14517</td>\n",
       "      <td>36</td>\n",
       "      <td>12635776</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2946788</td>\n",
       "      <td>C____302128_10</td>\n",
       "      <td>Reverse</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>TGAAAAGCTTTAGAAGAAAAAAGCT[G/T]TGTGGCTATTGAGTTT...</td>\n",
       "      <td>11</td>\n",
       "      <td>hCV302128</td>\n",
       "      <td>36</td>\n",
       "      <td>23967106</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3793451</td>\n",
       "      <td>C____320568_10</td>\n",
       "      <td>Reverse</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>GGTTATCATGGCTGCCCTCTCACTT[C/T]TTCAGAGACATGTGTT...</td>\n",
       "      <td>9</td>\n",
       "      <td>hCV320568</td>\n",
       "      <td>36</td>\n",
       "      <td>70849100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs10236187</td>\n",
       "      <td>C____328256_10</td>\n",
       "      <td>Forward</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>GAACGGCAGACAAAGCCTCACATTA[A/C]GCATCTCTTTAGTAAA...</td>\n",
       "      <td>7</td>\n",
       "      <td>hCV328256</td>\n",
       "      <td>36</td>\n",
       "      <td>139093846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs1569175</td>\n",
       "      <td>C____441412_10</td>\n",
       "      <td>Reverse</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TTCTTCCTCATCATCATCGAAGTTA[C/T]TTATTGATACCTCTTC...</td>\n",
       "      <td>2</td>\n",
       "      <td>hCV441412</td>\n",
       "      <td>36</td>\n",
       "      <td>200730199</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  NCBI SNP Reference        Assay ID  Strand  VIC FAM  \\\n",
       "0           rs731257  C_____14517_10  Forward   A   G   \n",
       "1          rs2946788  C____302128_10  Reverse   G   T   \n",
       "2          rs3793451  C____320568_10  Reverse   C   T   \n",
       "3         rs10236187  C____328256_10  Forward   A   C   \n",
       "4          rs1569175  C____441412_10  Reverse   C   T   \n",
       "\n",
       "                                    Context Sequence  Chr  Celera ID  \\\n",
       "0  AAGTTGCAATATGGCAAAACCTGTA[A/G]GAGATACAATTTGTGA...    7   hCV14517   \n",
       "1  TGAAAAGCTTTAGAAGAAAAAAGCT[G/T]TGTGGCTATTGAGTTT...   11  hCV302128   \n",
       "2  GGTTATCATGGCTGCCCTCTCACTT[C/T]TTCAGAGACATGTGTT...    9  hCV320568   \n",
       "3  GAACGGCAGACAAAGCCTCACATTA[A/C]GCATCTCTTTAGTAAA...    7  hCV328256   \n",
       "4  TTCTTCCTCATCATCATCGAAGTTA[C/T]TTATTGATACCTCTTC...    2  hCV441412   \n",
       "\n",
       "   NCBI Assembly Build Number   Location on NCBI Assembly  Unnamed: 10  \\\n",
       "0                           36                   12635776          NaN   \n",
       "1                           36                   23967106          NaN   \n",
       "2                           36                   70849100          NaN   \n",
       "3                           36                  139093846          NaN   \n",
       "4                           36                  200730199          NaN   \n",
       "\n",
       "   Unnamed: 11  Unnamed: 12  Unnamed: 13    1  \n",
       "0          NaN          NaN          NaN  143  \n",
       "1          NaN          NaN          NaN  167  \n",
       "2          NaN          NaN          NaN   38  \n",
       "3          NaN          NaN          NaN   91  \n",
       "4          NaN          NaN          NaN   50  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df128.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df128[['Chr', 'Location on NCBI Assembly']].to_csv('../data/Seldin_128_AISNPs.grch36.txt', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Remap from GRCh36 -> GRCh37  \n",
    "You can use [this tool](https://www.ncbi.nlm.nih.gov/genome/tools/remap) and `Seldin_128_AISNPs.grch36.txt` (exported in previous cell) to remap from GRChr36 to GRChr37. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "...converting to GRCh37..."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read the converted Seldin Lab 128 AISNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df128_hg19 = pd.read_csv('../data/report_Seldin_128_AISNPs.grch36.txt.xls', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#feat_name</th>\n",
       "      <th>source_int</th>\n",
       "      <th>mapped_int</th>\n",
       "      <th>source_id</th>\n",
       "      <th>mapped_id</th>\n",
       "      <th>source_length</th>\n",
       "      <th>mapped_length</th>\n",
       "      <th>source_start</th>\n",
       "      <th>source_stop</th>\n",
       "      <th>source_strand</th>\n",
       "      <th>source_sub_start</th>\n",
       "      <th>source_sub_stop</th>\n",
       "      <th>mapped_start</th>\n",
       "      <th>mapped_stop</th>\n",
       "      <th>mapped_strand</th>\n",
       "      <th>coverage</th>\n",
       "      <th>recip</th>\n",
       "      <th>asm_unit</th>\n",
       "      <th>Unnamed: 18</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Line:2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>12635776</td>\n",
       "      <td>12635776</td>\n",
       "      <td>+</td>\n",
       "      <td>12635776</td>\n",
       "      <td>12635776</td>\n",
       "      <td>12669251</td>\n",
       "      <td>12669251</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>First Pass</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Line:3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>23967106</td>\n",
       "      <td>23967106</td>\n",
       "      <td>+</td>\n",
       "      <td>23967106</td>\n",
       "      <td>23967106</td>\n",
       "      <td>24010530</td>\n",
       "      <td>24010530</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>First Pass</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Line:4</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>70849100</td>\n",
       "      <td>70849100</td>\n",
       "      <td>+</td>\n",
       "      <td>70849100</td>\n",
       "      <td>70849100</td>\n",
       "      <td>71659280</td>\n",
       "      <td>71659280</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>First Pass</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Line:5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>139093846</td>\n",
       "      <td>139093846</td>\n",
       "      <td>+</td>\n",
       "      <td>139093846</td>\n",
       "      <td>139093846</td>\n",
       "      <td>139447377</td>\n",
       "      <td>139447377</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>First Pass</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Line:6</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>200730199</td>\n",
       "      <td>200730199</td>\n",
       "      <td>+</td>\n",
       "      <td>200730199</td>\n",
       "      <td>200730199</td>\n",
       "      <td>201021954</td>\n",
       "      <td>201021954</td>\n",
       "      <td>+</td>\n",
       "      <td>1.0</td>\n",
       "      <td>First Pass</td>\n",
       "      <td>Primary Assembly</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #feat_name  source_int  mapped_int  source_id mapped_id  source_length  \\\n",
       "0     Line:2           1           1          7         7              1   \n",
       "1     Line:3           1           1         11        11              1   \n",
       "2     Line:4           1           1          9         9              1   \n",
       "3     Line:5           1           1          7         7              1   \n",
       "4     Line:6           1           1          2         2              1   \n",
       "\n",
       "   mapped_length  source_start  source_stop source_strand  source_sub_start  \\\n",
       "0              1      12635776     12635776             +          12635776   \n",
       "1              1      23967106     23967106             +          23967106   \n",
       "2              1      70849100     70849100             +          70849100   \n",
       "3              1     139093846    139093846             +         139093846   \n",
       "4              1     200730199    200730199             +         200730199   \n",
       "\n",
       "   source_sub_stop  mapped_start  mapped_stop mapped_strand  coverage  \\\n",
       "0         12635776      12669251     12669251             +       1.0   \n",
       "1         23967106      24010530     24010530             +       1.0   \n",
       "2         70849100      71659280     71659280             +       1.0   \n",
       "3        139093846     139447377    139447377             +       1.0   \n",
       "4        200730199     201021954    201021954             +       1.0   \n",
       "\n",
       "        recip          asm_unit  Unnamed: 18  \n",
       "0  First Pass  Primary Assembly          NaN  \n",
       "1  First Pass  Primary Assembly          NaN  \n",
       "2  First Pass  Primary Assembly          NaN  \n",
       "3  First Pass  Primary Assembly          NaN  \n",
       "4  First Pass  Primary Assembly          NaN  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df128_hg19.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write another new 1kG .vcf file, this time with the 128 Seldin AISNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# replace /Users/kevin/projects/tgviz with /path/to/1kg_bcfs\n",
    "bcf_path = '/Users/kevin/projects/tgviz/1kg_bcfs/'\n",
    "bcf_fname = 'ALL.chr{}.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.bcf'.format(1)\n",
    "bcf = VCF(os.path.join(bcf_path, bcf_fname))\n",
    "\n",
    "w = Writer('../data/Seldin.128AISNP.1kG.vcf', bcf)\n",
    "for i, aim in df128_hg19.iterrows():\n",
    "    chrom = aim['source_id']\n",
    "    pos = aim['mapped_start']\n",
    "    bcf_fname = 'ALL.chr{}.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.bcf'.format(chrom)\n",
    "\n",
    "    bcf = VCF(os.path.join(bcf_path, bcf_fname))\n",
    "    for variant in bcf(\"{}:{}-{}\".format(chrom, pos, pos)):\n",
    "        if variant.POS == int(pos):\n",
    "            w.write_record(variant)\n",
    "w.close(); bcf.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# There should be two `.vcf` files in the `data/` directory.\n",
    "\n",
    "* `data/Kidd.55AISNP.1kG.vcf` \n",
    "* `data/Seldin.128AISNP.1kG.vcf`"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:tgviz_env]",
   "language": "python",
   "name": "conda-env-tgviz_env-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
